sycl/include/sycl/atomic_ref.hpp

```diff
@@ -10,6 +10,7 @@
 
 #include <sycl/access/access.hpp> // for address_space
 #include <sycl/bit_cast.hpp>      // for bit_cast
+#include <sycl/half_type.hpp>     // for half
```

Comment:
Any chance we can make it work with forward declarations only? We do include the half/bfloat16 way too much already, but I'd much prefer if we stop spreading it further.

Reply:
Done [7eb1b5f](https://github.com/intel/llvm/pull/15158/commits/7eb1b5fab90a2f8efa61688195a882db95da0295)